public/app/features/dashboard/components/DashboardSettings/GeneralSettings.tsx

```diff
@@ -126,4 +133,15 @@ export const GeneralSettings: React.FC<Props> = ({ dashboard }) => {
       </div>
     </div>
   );
+}
+
+const mapDispatchToProps = {
+  updateTimeZone: (timeZone: TimeZone) => {
```

Comment:
Does this action cause `Error: Actions must be plain objects. Use custom middleware for async actions.` in console?

I think this either needs to be made into a proper ThunkAction as @hugohaggmark is suggesting or pass in the `updateTimeZoneForSession` as a prop and then move the body of this action inside `onTimeZoneChange`.

Reply:
Thank you Jack :), indeed it was causing an error in the console, I believed is fixed, I followed Hugo suggestion 787291e092d04907ebbc3db2a5824b8514f72af0